include/zephyr/devicetree.h

```diff
@@ -2243,14 +2243,28 @@
 #define DT_REG_HAS_NAME(node_id, name) \
 	IS_ENABLED(DT_CAT4(node_id, _REG_NAME_, name, _EXISTS))
 
+/**
+ * @cond INTERNAL_HIDDEN
+ */
+
+#define DT_REG_ADDR_BY_IDX_INTERNAL(node_id, idx) \
+	DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
+
+#define DT_REG_ADDR_INTERNAL(node_id) \
```

Comment:
I think usage information should part be doxygen comment, so it will be used correctly.

Would be nice if the name could reflect that it does not prepend a prefix and/or that it is used to get generated define AFAIK.

Reply:
yeah naming for this one could use more ideas, this should go for arch wg anyway imo, let me tag it so it gets blocked as well

Reply:
I'd like to see some documentation as well. How come this is internal but still used outside this file, for instance?